### DISCUSSION TITLE
Ubuntu，vscode报错

### DISCUSSION THREAD

**Discussion by ly-gump** [sjtug/SJTUThesis#903] [提问]
在vscode上使用Recipe：latexmk(xelatex)报错，貌似是参考文献的错误。输出如下：
[main.log](https://github.com/sjtug/SJTUThesis/files/11047043/main.log)

**Comment by LogCreative** [comment]
请提供完整的日志信息。

**Comment by ly-gump** [comment]
是这个log文件吗？我的vscode没有添加其他网上的一些setting.json文件，这个需要添加吗？我之前添加了也是报错的，而且我没添加也能跑SJTU-beamer，我就没添加了。谢谢！


祝安！


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "sjtug/SJTUThesis"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月22日(星期三) 晚上10:22
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [sjtug/SJTUThesis] Ubuntu，vscode报错 (Discussion #903)





 
请提供完整的日志信息。

**Reply by LogCreative** [reply]
对 把log文件上传 或者把报错部分的信息（带Error的行）贴出来

**Reply by ly-gump** [reply]
好的，我上传log文件了。

**Comment by LogCreative** [comment]
```
./texmf/tex/latex/sjtutex/sjtuthesis.cls:42: Package sjtutex Error: Package "expl3" is too old.
(sjtutex)                
(sjtutex)                Please update an up-to-date version of the bundles
(sjtutex)                "l3kernel" and "l3packages" using your TeX package
(sjtutex)                manager or from CTAN.

Type <return> to continue.
 ...                                              
                                                  
l.42 ...nnn { sjtutex } { l3-too-old } { expl3 } }
```

请更新至最新的 TeX 发行版。

**Reply by ly-gump** [reply]
我把texlive更新到了2023还是报一样的错误。
![image](https://user-images.githubusercontent.com/65701581/227167180-cdf8a460-f66e-4d3f-8985-8a1df4a6fcf8.png)

**Reply by LogCreative** [reply]
可以考虑删除缓存后重新编译，如果是同样的问题，建议看一下有没有卸载原来的版本。

**Reply by ly-gump** [reply]
卸载掉原来的版本之后，编译成功了。谢谢！
